atd-vze/src/views/Tools/CreateCrashRecord.js

```diff
@@ -93,295 +139,237 @@ const CreateCrashRecord = ({ client }) => {
+      setFeedback({
+        title: "Error",
+        message: "Must have a valid Crash Timestamp.",
+      });
+      return false;
+    }
+    if (isFieldInvalid(primaryStreetName)) {
+      setFeedback({
+        title: "Error",
+        message: "Must have a valid Primary Address.",
+      });
+      return false;
+    }
 
-            unitObjects = unitObjects.concat(`{
-              unit_nbr: ${unitNumber},
-            unit_desc_id: ${Number(unit.unit_desc_id)},
-            cris_schema_version: "2023",
-            updated_by: "${userEmail}",
-            created_by: "${userEmail}",
-            people_cris: {
-              data: [
-                ${personObjects}
-              ]
-            }
-          }`);
-          });
+    const unitObjects = buildNestedObjects();
 
-          const INSERT_BULK = gql`
-            mutation bulkInsert(
-              $rpt_street_name: String
-              $rpt_sec_street_name: String
-              $case_id: String
-              $crash_timestamp: timestamptz
-              $updated_by: String
-              $created_by: String
-            ) {
-              insert_crashes_cris(
-                objects: [
-                  {
-                    rpt_street_name: $rpt_street_name
-                    rpt_sec_street_name: $rpt_sec_street_name
-                    case_id: $case_id
-                    rpt_city_id: 22
-                    crash_timestamp: $crash_timestamp
-                    updated_by: $updated_by
-                    created_by: $created_by
-                    cris_schema_version: "2023"
-                    is_temp_record: true
-                    units_cris: { data: [${unitObjects}] }
-                  }
-                ]
-              ) {
-                returning {
-                  id
-                }
-              }
-            }
-          `;
-          const fatalityCountSum = unitFormState.reduce(
-            (a, b) => a + Number(b.fatality_count),
-            0
-          );
-          const susSeriousInjuryCountSum = unitFormState.reduce(
-            (a, b) => a + Number(b.sus_serious_injry_cnt),
-            0
-          );
+    const INSERT_BULK = gql`
+      mutation bulkInsert($crash_data: crashes_cris_insert_input!) {
+        insert_crashes_cris(objects: [$crash_data]) {
+          returning {
+            id
+          }
+        }
+      }
+    `;
 
-          const crashVariables = {
-            rpt_street_name: primaryStreetName.toUpperCase(),
-            rpt_sec_street_name: secondaryStreetName.toUpperCase(),
+    client
+      .mutate({
+        mutation: INSERT_BULK,
+        variables: {
+          crash_data: {
+            rpt_street_name: primaryStreetName?.toUpperCase(),
+            rpt_sec_street_name: secondaryStreetName?.toUpperCase(),
             case_id: caseId,
-            crash_timestamp: crashTimestamp,
+            rpt_city_id: 22,
+            crash_timestamp: new Date(crashTimestamp).toISOString(),
```

Comment:
i tested this in Chrom and Safari and it works as expected 👍 🙏

Reply:
firefox works too :partying_face: